ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -160,12 +160,19 @@
   [context metadata]
   (let [{subscriber :SubscriberId
          email :EmailAddress} metadata
-        subscriber (if-not subscriber
-                     (api-core/get-user-id-from-token context)
-                     subscriber)
-        email (if-not email
-                (urs/get-user-email context subscriber)
-                email)]
+        subscriber (if subscriber
+                     subscriber
+                     (api-core/get-user-id-from-token context))
+        email (if email
+                email
+                (if subscriber
+                  (urs/get-user-email context subscriber)
+                  ;; An error is thrown here to handle the case in which a subscription with no
+                  ;; subscriber-id is supplied, AND there is no token in the headers. In that case,
```

Comment:
Change `In that case, subscriber would be nil, and we don't want to hit URS for info on username 'nil'` to something like: `We will never allow a guest user to CRUD a subscription, so return a 401 error here.`

Reply:
👍